### ISSUE TITLE
Bad idea

### ISSUE THREAD

**Issue by aristov** [ismamz/posthtml-alt-always#2]
👎 

The `alt` attribute should always be marked up accurately by the author of a document.
The "automatically adding an empty `alt` everywhere" approach just encourages the web content authors to forget about accessibility of images at all and forever.
Mark up your images with `alt` properly instead of using this tool, if you care a bit of the accessibility of your document.

https://www.w3.org/TR/html51/semantics-embedded-content.html#alt-text

**Comment by ismamz** [comment]
Of course, you're right. No one should use this tool as an accessibility solution for alt text. I assume that if you use this plugin, you care about accessibility and you know basic rules to make a web page accessible.

You can't avoid that people forget alt text. However, you can prevent people from leaving images without any alt text. If you set empty alt, you're saying that "this image should be ignored" and this should be the default behavior when you forget alternative text.

This is a good article about this: https://www.joedolson.com/2015/03/are-alt-attributes-required-always/

**Comment by revelt** [comment]
@aristov I agree that images should be marked up correctly upfront. However, this plugin (at least to me) is more like _an insurance_. Coding in professional environment, in a team, across different systems, throughout a large frame, managed by different people and with people of variable coding capability touching the code, it's not that simple task to ensure correctness.

IMHO, each coding requirement (like the rule that ALT attributes must be present) should be covered 3 ways:
* Linter (for example, linter should catch and show live warning that ALT attribute is missing)
* Precommit hooks (performing same checks, just in case linter is down or settings are messed up, which happens), ideally ran outside linter, via CLI-type apps or even shell commands such as `grep`
* Automated post-processing should be ran on top of the build, as part of a build step. It should be different from linter, in case linter is unavailable (Mac/Windows) or it's different side (ie build server).

This third step is satisfied by `posthtml-alt-always`, in my case.

Using a combo of 3 it's nearly impossible to bypass any code rule/requirement.

**Comment by aristov** [comment]
Ok, I don't care.
